### PR TITLE
fix: Feature names no longer end in "Feature"

### LINKED ISSUE #2047: "/feature disable FooFeature" should not end in Feature
See #2044.

### PR DESCRIPTION
**Pull request by SpaceMonkeyy86**
When using the /feature or /wynntils feature commands to enable or disable features, the names of the features no longer end with the word "Feature."

Example: "/feature enable FooFeature" -> "/feature enable Foo"

Resolves #2047

### CODE REVIEW

**Comment by Mcrtin** [comment]
I think this better to add "Feature" to the string in `FeatureManager#getFeatureFromString` if it is not already ending with it. That way you can still distinguish between features and overlays using the short name and you can still add "Feature" at the end and it will still be recognized. Also this doesn't work with overlays.

**Comment by TwinBuilderOne** [comment]
> I think this better to add "Feature" to the string in `FeatureManager#getFeatureFromString` if it is not already ending with it.

This should already work because `FeatureManager#getFeatureFromString` calls `Feature#getShortName`.

**Comment by SpaceMonkeyy86** [comment]
Changing `FeatureManager#getFeatureFromString` would make /feature stop working. What do you mean when you say it doesn't work with overlays?

File: common/src/main/java/com/wynntils/core/consumers/features/Feature.java [hunk 1]
```diff
@@ -41,7 +41,8 @@ public String getTranslatedDescription() {
     }
 
     public String getShortName() {
-        return this.getClass().getSimpleName();
+        String str = this.getClass().getSimpleName();
+        return str.endsWith("Feature") ? str.substring(0, str.length() - 7) : str;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/core/consumers/features/Feature.java]: String#replace("Feature", "")

**Comment by kristofbolyai** [comment]
`String#replaceAll` is not `String#replace`.

**Comment by Mcrtin** [comment]
in `Overlay` is also a method called `getShortName`. Although rarely used, it'd be more consistent when also changing that.

**Comment by SpaceMonkeyy86** [comment]
> `String#replaceAll` is not `String#replace`.

My bad...

File: common/src/main/java/com/wynntils/core/consumers/features/Feature.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ public String getTranslatedDescription() {
     }
 
     public String getShortName() {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/core/consumers/features/Feature.java:43]: Are you sure that changing this does not have any unwanted side effects?

Comment (Mcrtin) [reply common/src/main/java/com/wynntils/core/consumers/features/Feature.java:43]: I checked that a while ago and as far as I can see this method (getShortName) is only used by getFeatureByName (or getOverlayByName) and that is only used by commands but someone should definitely verify that.

Comment (SpaceMonkeyy86) [reply common/src/main/java/com/wynntils/core/consumers/features/Feature.java:43]: Yes, the only places this function is used are for /config and /feature and for logging.

**Review by magicus — APPROVED** [review]

**Pull request merged**